core/web/log_controller_test.go

```diff
@@ -0,0 +1,89 @@
+package web_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"testing"
+
+	"github.com/bmizerany/assert"
+	"github.com/smartcontractkit/chainlink/core/internal/cltest"
+	"github.com/smartcontractkit/chainlink/core/services/eth"
+	"github.com/smartcontractkit/chainlink/core/web"
+	"github.com/smartcontractkit/chainlink/core/web/presenters"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zapcore"
+)
+
+type testCase struct {
+	Description string
+	logLevel    string
+	logSql      *bool
+
+	expectedLogLevel zapcore.Level
+}
+
+func TestLogController_SetDebug(t *testing.T) {
+	t.Parallel()
+
+	rpcClient, gethClient, _, assertMocksCalled := cltest.NewEthMocksWithStartupAssertions(t)
+	defer assertMocksCalled()
+	app, cleanup := cltest.NewApplicationWithKey(t,
+		eth.NewClientWith(rpcClient, gethClient),
+	)
+	defer cleanup()
+	require.NoError(t, app.Start())
+	client := app.NewHTTPClient()
+
+	sqlTrue := true
+	sqlFalse := false
+	cases := []testCase{
+		{
+			Description:      "Set log level to debug",
+			logLevel:         "debug",
+			logSql:           nil,
+			expectedLogLevel: zapcore.DebugLevel,
+		},
+		{
+			Description:      "Set log level to info",
+			logLevel:         "info",
+			logSql:           nil,
+			expectedLogLevel: zapcore.InfoLevel,
+		},
+		{
+			Description:      "Set log level to info and log sql to true",
+			logLevel:         "info",
+			logSql:           &sqlTrue,
+			expectedLogLevel: zapcore.InfoLevel,
+		},
+		{
+			Description:      "Set log level to warn and log sql to false",
+			logLevel:         "warn",
+			logSql:           &sqlFalse,
+			expectedLogLevel: zapcore.WarnLevel,
+		},
+	}
+
+	for _, tc := range cases {
+		func() {
```

Comment:
why the function here?

Reply:
this was to create a context for the `defer cleanup()`, probably not necessary, i can remove if no issue seen